### PR TITLE
Change implementation from Java to Kotlin

### PR DESCRIPTION
**Pull request by wmontwe**
### Changed to Kotlin

* `TagHelper`
* `FhirService`
* `ImageResizer`
* `FileService`
* `NetworkConnectivityService`
* `CryptoService`

## Motivation and Context
Prepare for easier FHIR 4 support and project maintenance

## How Has This Been Tested?
All existing tests are verified

## Types of changes
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:
- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have updated the changelog accordingly.
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

File: sdk-core/src/main/java/care/data4life/sdk/FhirService.kt [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+import care.data4life.fhir.stu3.model.DomainResource
+import care.data4life.fhir.stu3.model.FhirElementFactory
+import care.data4life.sdk.lang.D4LException
+import io.reactivex.Single
+
+// TODO remove @JvmOverloads when RecordService and Client changed to Kotlin
+// TODO use of Single is not necessary as it's finalized with blockingGet()
+internal class FhirService @JvmOverloads constructor(
+        private val cryptoService: CryptoService,
+        private val parserFhir3: FhirParser<Any> = Fhir().createStu3Parser()
```
Comment (bitPogo) [inline sdk-core/src/main/java/care/data4life/sdk/FhirService.kt:32]: `fhir3Parser` reads a bit nicer, in my opinion.

File: sdk-core/src/main/java/care/data4life/sdk/tags/TagHelper.kt [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+            tagList.add(key + TAG_DELIMITER + value)
+        }
+        return tagList
+    }
+
+    @JvmStatic
+    fun convertToTagMap(tagList: List<String>): HashMap<String, String> {
+        val tags = HashMap<String, String>()
+        for (entry in tagList) {
+            val split: Array<String?> = entry.split(TAG_DELIMITER.toRegex()).toTypedArray()
```
Comment (bitPogo) [inline sdk-core/src/main/java/care/data4life/sdk/tags/TagHelper.kt:37]: ~Array<String?>~

File: sdk-core/src/main/java/care/data4life/sdk/tags/TagHelper.kt [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+    private const val TAG_DELIMITER = "="
+
+    @JvmStatic
+    fun convertToTagList(tags: HashMap<String, String>): List<String> {
+        val tagList: MutableList<String> = ArrayList()
+        for ((key, value) in tags) {
+            tagList.add(key + TAG_DELIMITER + value)
+        }
+        return tagList
+    }
```
Comment (bitPogo) [inline sdk-core/src/main/java/care/data4life/sdk/tags/TagHelper.kt:31]: Just as a matter of taste:
``` kotlin
fun convertToTagList(
    tags: HashMap<String, String>
): List<String> = arrayListOf().also {
        for ((key, value) in tags) {
            it.add(key + TAG_DELIMITER + value)
        }
}
```

File: sdk-core/src/main/java/care/data4life/sdk/FhirService.kt [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+) {
+
+    @Suppress("UNCHECKED_CAST")
+    fun <T : DomainResource> decryptResource(dataKey: GCKey, resourceType: String, encryptedResource: String): T {
+        return Single
+                .just(encryptedResource)
+                .filter { encResource: String -> !encResource.isBlank() }
```
Comment (bitPogo) [inline sdk-core/src/main/java/care/data4life/sdk/FhirService.kt:39]: `!encResource.isBlank()`-> `encResource.isNotBlank()`

File: sdk-core/src/main/java/care/data4life/sdk/FhirService.kt [hunk 5]
```diff
@@ -0,0 +1,64 @@
...
+                .map { encResource: String -> cryptoService.decryptString(dataKey, encResource).blockingGet() }
```
Comment (bitPogo) [inline sdk-core/src/main/java/care/data4life/sdk/FhirService.kt:40]: : ~String~

File: sdk-core/src/main/java/care/data4life/sdk/FhirService.kt (hunk 4 above)
Comment (bitPogo) [inline sdk-core/src/main/java/care/data4life/sdk/FhirService.kt:39]: : ~String~

File: sdk-core/src/main/java/care/data4life/sdk/FhirService.kt [hunk 6]
```diff
@@ -0,0 +1,64 @@
...
+                    Single.error(
+                            DecryptionFailed("Failed to decrypt resource", error) as D4LException)
+                }
+                .blockingGet() as T
+    }
+
+    fun <T : DomainResource> encryptResource(dataKey: GCKey, resource: T): String {
+        return Single
+                .just(resource)
+                .map { fhirObject: T -> parserFhir3.fromFhir(fhirObject) }
```
Comment (bitPogo) [inline sdk-core/src/main/java/care/data4life/sdk/FhirService.kt:55]: ~: T~

File: sdk-core/src/main/java/care/data4life/sdk/FhirService.kt [hunk 7]
```diff
@@ -0,0 +1,64 @@
...
+                .flatMap { json: String -> cryptoService.encryptString(dataKey, json) }
```
Comment (bitPogo) [inline sdk-core/src/main/java/care/data4life/sdk/FhirService.kt:56]: ~: String~

File: sdk-core/src/main/java/care/data4life/sdk/FileService.kt [hunk 8]
```diff
@@ -0,0 +1,45 @@
...
+internal class FileService(
+        private val alias: String,
+        private val apiService: ApiService,
+        private val cryptoService: CryptoService
+) {
+
+    fun downloadFile(key: GCKey, userId: String, fileId: String): Single<ByteArray> {
+        return apiService
+                .downloadDocument(alias, userId, fileId)
+                .flatMap { downloadedFile: ByteArray -> cryptoService.decrypt(key, downloadedFile) }
```
Comment (bitPogo) [inline sdk-core/src/main/java/care/data4life/sdk/FileService.kt:31]: ~: ByteArray~

File: sdk-core/src/main/java/care/data4life/sdk/FileService.kt [hunk 9]
```diff
@@ -0,0 +1,45 @@
...
+                .onErrorResumeNext { error -> Single.error(FileException.DownloadFailed(error)) }
+    }
+
+    fun uploadFile(key: GCKey, userId: String, data: ByteArray): Single<String> {
+        return cryptoService
+                .encrypt(key, data)
+                .flatMap { encryptedData: ByteArray -> apiService.uploadDocument(alias, userId, encryptedData) }
```
Comment (bitPogo) [inline sdk-core/src/main/java/care/data4life/sdk/FileService.kt:38]: ~: ByteArray~

File: sdk-core/src/main/java/care/data4life/sdk/CryptoService.kt [hunk 10]
```diff
@@ -0,0 +1,323 @@
...
+                .onErrorResumeNext { error ->
+                    Log.error(error, "Failed to encrypt string")
+                    Single.error(CryptoException.EncryptionFailed("Failed to encrypt string") as D4LException)
+                }
+    }
+
+    fun decryptString(key: GCKey, dataBase64: String): Single<String> {
+        return Single
+                .fromCallable { base64.decode(dataBase64) }
+                .flatMap { decoded: ByteArray -> decrypt(key, decoded) }
```
Comment (bitPogo) [inline sdk-core/src/main/java/care/data4life/sdk/CryptoService.kt:130]: ~: ByteArray~

File: sdk-core/src/main/java/care/data4life/sdk/CryptoService.kt [hunk 11]
```diff
@@ -0,0 +1,323 @@
...
+                .map { decrypted: ByteArray -> String(decrypted, Charset.forName("UTF-8")) }
```
Comment (bitPogo) [inline sdk-core/src/main/java/care/data4life/sdk/CryptoService.kt:131]: ~: ByteArray~

Comment (bitPogo) [inline sdk-core/src/main/java/care/data4life/sdk/CryptoService.kt:131]: This really discussible: 
"UTF-8" -> StandardCharsets.UTF_8.name()

File: sdk-core/src/main/java/care/data4life/sdk/CryptoService.kt [hunk 12]
```diff
@@ -0,0 +1,323 @@
...
+        return decryptString(key, encryptedKey.encryptedKey)
+                .map { keyJson -> moshi.adapter(ExchangeKey::class.java).fromJson(keyJson) }
+                .flatMap { exchangeKey -> convertExchangeKeyToGCKey(exchangeKey) }
+    }
+
+    fun asymDecryptSymetricKey(keyPair: GCKeyPair, encryptedKey: EncryptedKey): Single<GCKey> {
+        return Single.fromCallable { base64.decode(encryptedKey.encryptedKey) }
+                .map { decoded -> asymDecrypt(keyPair, decoded) }
+                .map { keyJson -> moshi.adapter(ExchangeKey::class.java).fromJson(String(keyJson, Charset.forName("UTF-8"))) }
+                .flatMap { exchangeKey -> convertExchangeKeyToGCKey(exchangeKey) }
```
Comment (bitPogo) [inline sdk-core/src/main/java/care/data4life/sdk/CryptoService.kt:159]: This really discussible: 
"UTF-8" -> StandardCharsets.UTF_8.name()

File: sdk-core/src/main/java/care/data4life/sdk/CryptoService.kt [hunk 13]
```diff
@@ -0,0 +1,323 @@
...
+                    Log.error(error, "Failed to decrypt string")
+                    Single.error(CryptoException.DecryptionFailed("Failed to decrypt string") as D4LException)
+                }
+    }
+
+    fun encryptSymmetricKey(key: GCKey, keyType: KeyType, gckey: GCKey): Single<EncryptedKey> {
+        return Single.fromCallable { createKey(KEY_VERSION, keyType, gckey.getKeyBase64()) }
+                .map { exchangeKey -> moshi.adapter(ExchangeKey::class.java).toJson(exchangeKey) }
+                .flatMap { jsonKey -> encryptString(key, jsonKey) }
+                .map<EncryptedKey> { encryptedKeyBase64 -> EncryptedKey(encryptedKeyBase64) }
```
Comment (bitPogo) [inline sdk-core/src/main/java/care/data4life/sdk/CryptoService.kt:142]: ~\<EncryptedKey\>~

File: sdk-core/src/main/java/care/data4life/sdk/CryptoService.kt [hunk 14]
```diff
@@ -0,0 +1,323 @@
...
+                    }
+                    Single.error(CryptoException.KeyDecryptionFailed("Failed to decrypt exchange key") as D4LException)
+                }
+    }
+
+    fun convertAsymmetricKeyToBase64ExchangeKey(gcAsymmetricKey: GCAsymmetricKey): Single<String> {
+        return Single.fromCallable { gcAsymmetricKey.value.encoded } //SPKI
+                .map(base64::encodeToString)
+                .map { encodedKey -> createKey(KEY_VERSION, KeyType.APP_PUBLIC_KEY, encodedKey) }
+                .map { exchangeKey: ExchangeKey -> moshi.adapter(ExchangeKey::class.java).toJson(exchangeKey) }
```
Comment (bitPogo) [inline sdk-core/src/main/java/care/data4life/sdk/CryptoService.kt:185]: ~: ExchangeKey~

File: sdk-core/src/main/java/care/data4life/sdk/CryptoService.kt [hunk 15]
```diff
@@ -0,0 +1,323 @@
...
+                .map { data: String -> base64.encodeToString(data) }
```
Comment (bitPogo) [inline sdk-core/src/main/java/care/data4life/sdk/CryptoService.kt:186]: ~: String~

File: sdk-core/src/main/java/care/data4life/sdk/CryptoService.kt [hunk 16]
```diff
@@ -0,0 +1,323 @@
...
+                    Single.error(CryptoException.KeyFetchingFailed("Failed to fetch encryption key") as D4LException)
+                }
+    }
+
+    fun storeTagEncryptionKey(tek: GCKey) {
+        storeKey(TEK_KEY, tek, KeyType.TAG_KEY)
+    }
+
+    @Throws(IOException::class)
+    fun fetchTagEncryptionKey(): GCKey {
```
Comment (bitPogo) [inline sdk-core/src/main/java/care/data4life/sdk/CryptoService.kt:265]: fun fetchTagEncryptionKey(): GCKey = getGCKey(TEK_KEY)

File: sdk-core/src/main/java/care/data4life/sdk/CryptoService.kt [hunk 17]
```diff
@@ -0,0 +1,323 @@
...
+    }
+
+    private fun storeKey(key: String, value: GCKey, keyType: KeyType) {
+        storage.storeKey(prefix() + key, value, keyType)
+    }
+
+    @Throws(IOException::class)
+    private fun getGCKey(key: String): GCKey {
+        return keyFactory.createGCKey(getExchangeKey(key))
+    }
```
Comment (bitPogo) [inline sdk-core/src/main/java/care/data4life/sdk/CryptoService.kt:276]: @Throws(IOException::class)
private fun getGCKey(key: String): GCKey = keyFactory.createGCKey(getExchangeKey(key))

File: sdk-core/src/main/java/care/data4life/sdk/CryptoService.kt [hunk 18]
```diff
@@ -0,0 +1,323 @@
...
+
+    @Throws(IOException::class)
+    private fun getExchangeKey(key: String): ExchangeKey {
+        return storage.getExchangeKey(prefix() + key)
+    }
```
Comment (bitPogo) [inline sdk-core/src/main/java/care/data4life/sdk/CryptoService.kt:281]: @Throws(IOException::class)
private fun getGCKey(key: String): GCKey = keyFactory.createGCKey(getExchangeKey(key))

File: sdk-core/src/main/java/care/data4life/sdk/CryptoService.kt [hunk 19]
```diff
@@ -0,0 +1,323 @@
...
+
+    private fun deleteSecret(key: String) {
+        storage.deleteSecret(prefix() + key)
+    }
+
+    private fun prefix(): String {
+        return alias + "_"
+    }
```
Comment (bitPogo) [inline sdk-core/src/main/java/care/data4life/sdk/CryptoService.kt:289]: private fun prefix(): String = alias + "_"

File: sdk-core/src/main/java/care/data4life/sdk/CryptoService.kt [hunk 20]
```diff
@@ -0,0 +1,323 @@
...
+
+    // Common Key Handling
+    @Throws(IOException::class)
+    fun fetchCurrentCommonKey(): GCKey {
+        return commonKeyService.fetchCurrentCommonKey()
+    }
```
Comment (bitPogo) [inline sdk-core/src/main/java/care/data4life/sdk/CryptoService.kt:295]: @Throws(IOException::class)
fun fetchCurrentCommonKey(): GCKey = commonKeyService.fetchCurrentCommonKey()

File: sdk-core/src/main/java/care/data4life/sdk/CryptoService.kt [hunk 21]
```diff
@@ -0,0 +1,323 @@
...
+
+    @Throws(IOException::class)
+    fun getCommonKeyById(commonKeyId: String): GCKey {
+        return commonKeyService.fetchCommonKey(commonKeyId)
+    }
```
Comment (bitPogo) [inline sdk-core/src/main/java/care/data4life/sdk/CryptoService.kt:300]: @Throws(IOException::class)
fun getCommonKeyById(commonKeyId: String): GCKey = commonKeyService.fetchCommonKey(commonKeyId)

File: sdk-core/src/main/java/care/data4life/sdk/CryptoService.kt [hunk 22]
```diff
@@ -0,0 +1,323 @@
...
+        commonKeyService.storeCurrentCommonKeyId(commonKeyId)
+    }
+
+    fun storeCommonKey(commonKeyId: String, commonKey: GCKey) {
+        commonKeyService.storeCommonKey(commonKeyId, commonKey)
+    }
+
+    fun hasCommonKey(commonKeyId: String): Boolean {
+        return commonKeyService.hasCommonKey(commonKeyId)
+    }
```
Comment (bitPogo) [inline sdk-core/src/main/java/care/data4life/sdk/CryptoService.kt:315]: fun hasCommonKey(commonKeyId: String): Boolean = commonKeyService.hasCommonKey(commonKeyId)

File: sdk-core/src/main/java/care/data4life/sdk/FhirService.kt (hunk 4 above)
Comment (bitPogo) [reply sdk-core/src/main/java/care/data4life/sdk/FhirService.kt:39]: ~: String~

File: sdk-core/src/main/java/care/data4life/sdk/FhirService.kt (hunk 5 above)
Comment (bitPogo) [inline sdk-core/src/main/java/care/data4life/sdk/FhirService.kt:40]: ~: String~

File: sdk-core/src/main/java/care/data4life/sdk/FhirService.kt [hunk 23]
```diff
@@ -0,0 +1,64 @@
...
+        private val parserFhir3: FhirParser<Any> = Fhir().createStu3Parser()
+) {
+
+    @Suppress("UNCHECKED_CAST")
+    fun <T : DomainResource> decryptResource(dataKey: GCKey, resourceType: String, encryptedResource: String): T {
+        return Single
+                .just(encryptedResource)
+                .filter { encResource: String -> !encResource.isBlank() }
+                .map { encResource: String -> cryptoService.decryptString(dataKey, encResource).blockingGet() }
+                .map<Any> { decryptedResourceJson: String ->
```
Comment (bitPogo) [inline sdk-core/src/main/java/care/data4life/sdk/FhirService.kt:41]: ~: String~

File: sdk-core/src/main/java/care/data4life/sdk/FhirService.kt (hunk 6 above)
Comment (bitPogo) [inline sdk-core/src/main/java/care/data4life/sdk/FhirService.kt:55]: ~: T~

File: sdk-core/src/main/java/care/data4life/sdk/FhirService.kt (hunk 7 above)
Comment (bitPogo) [inline sdk-core/src/main/java/care/data4life/sdk/FhirService.kt:56]: ~: String~

File: sdk-core/src/main/java/care/data4life/sdk/FileService.kt (hunk 8 above)
Comment (bitPogo) [inline sdk-core/src/main/java/care/data4life/sdk/FileService.kt:31]: ~: ByteArray~

File: sdk-core/src/main/java/care/data4life/sdk/FileService.kt (hunk 9 above)
Comment (bitPogo) [inline sdk-core/src/main/java/care/data4life/sdk/FileService.kt:38]: ~: ByteArray~

File: sdk-core/src/main/java/care/data4life/sdk/FileService.kt [hunk 24]
```diff
@@ -0,0 +1,45 @@
...
+                .onErrorResumeNext { error -> Single.error(FileException.UploadFailed(error)) }
+    }
+
+    fun deleteFile(userId: String, fileId: String): Single<Boolean> {
+        return apiService.deleteDocument(alias, userId, fileId)
+    }
```
Comment (bitPogo) [inline sdk-core/src/main/java/care/data4life/sdk/FileService.kt:44]: fun deleteFile(userId: String, fileId: String): Single<Boolean> = apiService.deleteDocument(alias, userId, fileId)

File: sdk-core/src/main/java/care/data4life/sdk/tags/TagHelper.kt (hunk 2 above)
Comment (bitPogo) [inline sdk-core/src/main/java/care/data4life/sdk/tags/TagHelper.kt:37]: `val split = entry.split(TAG_DELIMITER)` should be sufficient

**Review by bitPogo — COMMENTED** [review]
Just nitpicks

File: sdk-core/src/main/java/care/data4life/sdk/tags/TagHelper.kt (hunk 3 above)
Comment (wmontwe) [reply sdk-core/src/main/java/care/data4life/sdk/tags/TagHelper.kt:31]: I move the return

```
fun convertToTagList(tags: HashMap<String, String>): List<String> {
        return mutableListOf<String>().also {
            for ((key, value) in tags) {
                it.add(key + TAG_DELIMITER + value)
            }
        }
}
```

File: sdk-core/src/main/java/care/data4life/sdk/FhirService.kt (hunk 4 above)
Comment (wmontwe) [reply sdk-core/src/main/java/care/data4life/sdk/FhirService.kt:39]: I didn't address that one as I refactored parts already in the FHIR4 branch

File: sdk-core/src/main/java/care/data4life/sdk/FhirService.kt (hunk 1 above)
Comment (wmontwe) [reply sdk-core/src/main/java/care/data4life/sdk/FhirService.kt:32]: I work on that in the FHIR 4 branch

File: sdk-core/src/main/java/care/data4life/sdk/FhirService.kt (hunk 5 above)
Comment (wmontwe) [reply sdk-core/src/main/java/care/data4life/sdk/FhirService.kt:40]: -> FHIR 4 branch

File: sdk-core/src/main/java/care/data4life/sdk/FhirService.kt (hunk 6 above)
Comment (wmontwe) [reply sdk-core/src/main/java/care/data4life/sdk/FhirService.kt:55]: -> FHIR 4 branch

File: sdk-core/src/main/java/care/data4life/sdk/FhirService.kt (hunk 7 above)
Comment (wmontwe) [reply sdk-core/src/main/java/care/data4life/sdk/FhirService.kt:56]: -> FHIR 4 branch

File: sdk-core/src/main/java/care/data4life/sdk/CryptoService.kt (hunk 11 above)
Comment (wmontwe) [reply sdk-core/src/main/java/care/data4life/sdk/CryptoService.kt:131]: Didn't know that this exists. Found also a nice replacement from Kotlin `Charsets.UTF_8`

File: sdk-core/src/main/java/care/data4life/sdk/FhirService.kt (hunk 5 above)
Comment (wmontwe) [reply sdk-core/src/main/java/care/data4life/sdk/FhirService.kt:40]: FHIR 4 branch

File: sdk-core/src/main/java/care/data4life/sdk/FhirService.kt (hunk 23 above)
Comment (wmontwe) [reply sdk-core/src/main/java/care/data4life/sdk/FhirService.kt:41]: FHIR 4 branch

File: sdk-core/src/main/java/care/data4life/sdk/FhirService.kt (hunk 6 above)
Comment (wmontwe) [reply sdk-core/src/main/java/care/data4life/sdk/FhirService.kt:55]: FHIR 4 branch

File: sdk-core/src/main/java/care/data4life/sdk/FhirService.kt (hunk 7 above)
Comment (wmontwe) [reply sdk-core/src/main/java/care/data4life/sdk/FhirService.kt:56]: FHIR 4 branch

File: sdk-core/src/main/java/care/data4life/sdk/FileService.kt (hunk 8 above)
Comment (wmontwe) [reply sdk-core/src/main/java/care/data4life/sdk/FileService.kt:31]: FHIR 4 branch

File: sdk-core/src/main/java/care/data4life/sdk/FileService.kt (hunk 24 above)
Comment (wmontwe) [reply sdk-core/src/main/java/care/data4life/sdk/FileService.kt:44]: I agree on oneliners using `=` but this requires a new line and it's harder to read. So I would leave it as is

File: sdk-core/src/main/java/care/data4life/sdk/tags/TagHelper.kt (hunk 2 above)
Comment (wmontwe) [reply sdk-core/src/main/java/care/data4life/sdk/tags/TagHelper.kt:37]: comment is doubled

File: sdk-core/src/main/java/care/data4life/sdk/FhirService.kt (hunk 1 above)
Comment (bitPogo) [reply sdk-core/src/main/java/care/data4life/sdk/FhirService.kt:32]: 👍

File: sdk-core/src/main/java/care/data4life/sdk/FhirService.kt (hunk 4 above)
Comment (bitPogo) [reply sdk-core/src/main/java/care/data4life/sdk/FhirService.kt:39]: 👍

Comment (bitPogo) [reply sdk-core/src/main/java/care/data4life/sdk/FhirService.kt:39]: 👍

File: sdk-core/src/main/java/care/data4life/sdk/FhirService.kt (hunk 5 above)
Comment (bitPogo) [reply sdk-core/src/main/java/care/data4life/sdk/FhirService.kt:40]: 👍

Comment (bitPogo) [reply sdk-core/src/main/java/care/data4life/sdk/FhirService.kt:40]: 👍

File: sdk-core/src/main/java/care/data4life/sdk/FhirService.kt (hunk 23 above)
Comment (bitPogo) [reply sdk-core/src/main/java/care/data4life/sdk/FhirService.kt:41]: 👍

File: sdk-core/src/main/java/care/data4life/sdk/FhirService.kt (hunk 6 above)
Comment (bitPogo) [reply sdk-core/src/main/java/care/data4life/sdk/FhirService.kt:55]: 👍

Comment (bitPogo) [reply sdk-core/src/main/java/care/data4life/sdk/FhirService.kt:55]: 👍

File: sdk-core/src/main/java/care/data4life/sdk/FhirService.kt (hunk 7 above)
Comment (bitPogo) [reply sdk-core/src/main/java/care/data4life/sdk/FhirService.kt:56]: 👍

Comment (bitPogo) [reply sdk-core/src/main/java/care/data4life/sdk/FhirService.kt:56]: 👍

File: sdk-core/src/main/java/care/data4life/sdk/FileService.kt (hunk 24 above)
Comment (bitPogo) [reply sdk-core/src/main/java/care/data4life/sdk/FileService.kt:44]: 👍

File: sdk-core/src/main/java/care/data4life/sdk/tags/TagHelper.kt (hunk 3 above)
Comment (bitPogo) [reply sdk-core/src/main/java/care/data4life/sdk/tags/TagHelper.kt]: Did you changed it? I looks like before? However, I am also fine with the version no-also version, but I am not sure, which you intended to use actually.

File: sdk-core/src/main/java/care/data4life/sdk/tags/TagHelper.kt (hunk 2 above)
Comment (bitPogo) [reply sdk-core/src/main/java/care/data4life/sdk/tags/TagHelper.kt]: What do you mean...that there are two comments on this line? Or this comment - in this case I see it only once.

**Review by bitPogo — COMMENTED** [review]
Just respond to the TagHelper comments and I will approve this PR.

**Comment by wmontwe** [comment]
@bitPogo the GitHub Review feature is currently broken, please have a look at the change commit itself https://github.com/d4l-data4life/hc-sdk-kmp/pull/21/commits/d4149c97aed2fe26638a714a74c997b3f8eb4184

**Review by bitPogo — APPROVED** [review]
Looks good to me

**Pull request merged**